### ISSUE TITLE
aws/s3_bucket: Expose region + hosted zone ID

### ISSUE THREAD

**Issue by radeksimko** [hashicorp/terraform#1769]
This is likely related to https://github.com/hashicorp/terraform/pull/1738, it can either be done in that PR, or separately once that PR is merged.

Some resources (probably should go into code as a hard-coded map):
http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_website_region_endpoints

This will be necessary for managing Route 53 ALIAS records pointing to S3 buckets.

cc @justincampbell

**Comment by justincampbell** [comment]
@radeksimko would the `website_endpoint` attribute that #1738 adds be sufficient, instead of exporting the region?

The zone ID is interesting.

**Comment by radeksimko** [comment]
@justincampbell Basically I think that the best way to approach the problem would be to change [`websiteEndpoint(s3conn *s3.S3, d *schema.ResourceData)`](https://github.com/hashicorp/terraform/pull/1738/files#diff-25502ab732bf1bacf47160e740e0f9b1R220) to `getWebsiteEndpointUrl(bucket, region string)`, i.e. refactor all location/region related logic outside of this function, then you can create a new function, e.g. `getHostedZoneId(region string)` which won't do anything more complicated than just map lookup and finally export the output.

**Comment by radeksimko** [comment]
And in that case ^ region may not be necessary, but I would export it anyway, if we allow creating buckets in different regions in the future (not just in the region coming from provider).

**Comment by justincampbell** [comment]
:+1: I'll either do it in that PR or open a new right after, within a few days

**Comment by justincampbell** [comment]
Done in https://github.com/hashicorp/terraform/compare/master...justincampbell:s3-region-zone

Either need to wait for feedback on https://github.com/awslabs/aws-sdk-go/issues/222, or add it to Terraform temporarily.

**Comment by justincampbell** [comment]
@radeksimko https://github.com/awslabs/aws-sdk-go/pull/222 was rejected. I don't think this logic belongs in TF, but I'd like others to chime in.

**Comment by radeksimko** [comment]
Hm, that sucks... but I still believe that letting user looking for this in the official AWS documentation is a bad approach.

**Comment by radeksimko** [comment]
Worst case scenario, we can at least link to the documentation from the route53_record ALIAS section.

**Comment by phinze** [comment]
While I agree it doesn't ultimately belong in TF, I'd toss my vote in for UX over cleanliness here and say we include the metadata. Terraform already holds knowledge of things like historically common config env variables for user benefit, and I don't think it's too much of a stretch to consider this mapping to be in the same arena.

That being said, if it's metadata we expect to change more than, say, once every 1-2 years, then perhaps we throw our hands up.
